test_scipy_constraints_2 from test_scipy fails

When building rpms for iminuit for openSUSE Tumbleweed, we run the tests to ensure the build is ok. With version 2.15.2, we see that the scipy test fails with the following error from `tests/test_scipy.py:216`(full failure log from pytest pasted at the end of this report):

```
[   36s] >                       A = np.atleast_1d(c.A)[:, cfree]
[   36s] E                       IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed
```

The following versions of libraries were used for the build/tests:

* Python 3.8.13
* Scipy 1.8.1
* Numpy 1.21.6

`pytest` failures report:

```
[   36s] =================================== FAILURES ===================================
[   36s] ________________________ test_scipy_constraints_2[True] ________________________
[   36s] 
[   36s] fixed = True
[   36s] 
[   36s]     @pytest.mark.parametrize("fixed", (False, True))
[   36s]     def test_scipy_constraints_2(fixed):
[   36s]         def fcn(x, y):
[   36s]             return x**2 + y**2
[   36s]     
[   36s]         m = Minuit(fcn, x=1, y=2)
[   36s]         m.fixed["x"] = fixed
[   36s]         con = scopt.LinearConstraint([1, 1], 0.1, np.inf)
[   36s] >       m.scipy(method="COBYLA", constraints=con)
[   36s] 
[   36s] tests/test_scipy.py:216: 
[   36s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   36s] 
[   36s] self = (Param(number=0, name='x', value=1.0, error=0.01, merror=None, is_const=False, is_fixed=True, lower_limit=None, upper_...r=1, name='y', value=2.0, error=0.02, merror=None, is_const=False, is_fixed=False, lower_limit=None, upper_limit=None))
[   36s] method = 'COBYLA', ncall = 305, hess = None, hessp = None
[   36s] constraints = [<scipy.optimize._constraints.LinearConstraint object at 0x7f5f9f1d8700>]
[   36s] 
[   36s]     def scipy(
[   36s]         self,
[   36s]         method: _tp.Optional[_tp.Union[str, _tp.Callable]] = None,
[   36s]         ncall: _tp.Optional[int] = None,
[   36s]         hess: _tp.Optional[_tp.Any] = None,
[   36s]         hessp: _tp.Optional[_tp.Any] = None,
[   36s]         constraints: _tp.Optional[_tp.Iterable[_tp.Any]] = None,
[   36s]     ) -> "Minuit":
[   36s]         """
[   36s]         Minimize with SciPy algorithms.
[   36s]     
[   36s]         Parameters
[   36s]         ----------
[   36s]         method : str or Callable, optional
[   36s]             Which scipy method to use.
[   36s]         ncall : int, optional
[   36s]             Function call limit.
[   36s]         hess : Callable, optional
[   36s]             Function that computes the Hessian matrix. It must use the exact same
[   36s]             calling conversion as the original fcn (several arguments which are numbers
[   36s]             or a single array argument).
[   36s]         hessp : Callable, optional
[   36s]             Function that computes the product of the Hessian matrix with a vector.
[   36s]             It must use the same calling conversion as the original fcn (several
[   36s]             arguments which are numbers or a single array argument) and end with another
[   36s]             argument which is an arbitrary vector.
[   36s]         constraints : scipy.optimize.LinearConstraint or
[   36s]                       scipy.optimize.NonlinearConstraint, optional
[   36s]             Linear or non-linear constraints, see docs of :func:`scipy.optimize.minimize`
[   36s]             look for the `constraints` parameter. The function used in the constraint
[   36s]             must use the exact same calling convention as the original fcn, see hess
[   36s]             parameter for details. No parameters may be omitted in the signature, even
[   36s]             if those parameters are not used in the constraint.
[   36s]     
[   36s]         Notes
[   36s]         -----
[   36s]         The call limit may be violated since many algorithms checks the call limit only
[   36s]         after a full iteraction of their algorithm, which consists of several function
[   36s]         calls. Some algorithms do not check the number of function calls at all, in this
[   36s]         case the call limit acts on the number of iterations of the algorithm. This
[   36s]         issue should be fixed in scipy.
[   36s]     
[   36s]         The SciPy minimizers use their own internal rule for convergence. The EDM
[   36s]         criterion is evaluated only after the original algorithm already stopped. This
[   36s]         means that usually SciPy minimizers will use more iterations than Migrad and
[   36s]         the tolerance :attr:`tol` has no effect on SciPy minimizers.
[   36s]         """
[   36s]         try:
[   36s]             from scipy.optimize import (
[   36s]                 minimize,
[   36s]                 Bounds,
[   36s]                 NonlinearConstraint,
[   36s]                 LinearConstraint,
[   36s]             )
[   36s]         except ImportError as exc:
[   36s]             exc.msg += "\n\nPlease install scipy to use scipy minimizers in iminuit."
[   36s]             raise
[   36s]     
[   36s]         if ncall is None:
[   36s]             ncall = self._migrad_maxcall()
[   36s]     
[   36s]         cfree = ~np.array(self.fixed[:], dtype=bool)
[   36s]         cpar = np.array(self.values[:])
[   36s]         no_fixed_parameters = self.nfit == self.npar
[   36s]     
[   36s]         if no_fixed_parameters:
[   36s]     
[   36s]             class Wrapped:
[   36s]                 __slots__ = ("fcn",)
[   36s]     
[   36s]                 def __init__(self, fcn):
[   36s]                     self.fcn = fcn
[   36s]     
[   36s]                 if self.fcn._array_call:
[   36s]     
[   36s]                     def __call__(self, par):
[   36s]                         return self.fcn(par)
[   36s]     
[   36s]                 else:
[   36s]     
[   36s]                     def __call__(self, par):
[   36s]                         return self.fcn(*par)
[   36s]     
[   36s]             WrappedGrad = Wrapped
[   36s]             WrappedHess = Wrapped
[   36s]     
[   36s]             class WrappedHessp:
[   36s]                 __slots__ = ("fcn",)
[   36s]     
[   36s]                 def __init__(self, fcn):
[   36s]                     self.fcn = fcn
[   36s]     
[   36s]                 if self.fcn._array_call:
[   36s]     
[   36s]                     def __call__(self, par, v):
[   36s]                         return self.fcn(par, v)
[   36s]     
[   36s]                 else:
[   36s]     
[   36s]                     def __call__(self, par, v):
[   36s]                         return self.fcn(*par, v)
[   36s]     
[   36s]         else:
[   36s]     
[   36s]             class Wrapped:  # type:ignore
[   36s]                 __slots__ = ("fcn", "free", "par")
[   36s]     
[   36s]                 def __init__(self, fcn):
[   36s]                     self.fcn = fcn
[   36s]                     self.free = cfree
[   36s]                     self.par = cpar
[   36s]     
[   36s]                 if self.fcn._array_call:
[   36s]     
[   36s]                     def __call__(self, par):
[   36s]                         self.par[self.free] = par
[   36s]                         return self.fcn(self.par)
[   36s]     
[   36s]                 else:
[   36s]     
[   36s]                     def __call__(self, par):
[   36s]                         self.par[self.free] = par
[   36s]                         return self.fcn(*self.par)
[   36s]     
[   36s]             class WrappedGrad(Wrapped):  # type:ignore
[   36s]                 def __call__(self, par):
[   36s]                     g = super().__call__(par)
[   36s]                     return np.atleast_1d(g)[self.free]
[   36s]     
[   36s]             class WrappedHess(Wrapped):  # type:ignore
[   36s]                 def __init__(self, fcn):
[   36s]                     super().__init__(fcn)
[   36s]                     self.freem = np.outer(self.free, self.free)
[   36s]                     n = np.sum(self.free)
[   36s]                     self.shape = n, n
[   36s]     
[   36s]                 def __call__(self, par):
[   36s]                     h = super().__call__(par)
[   36s]                     return np.atleast_2d(h)[self.freem].reshape(self.shape)
[   36s]     
[   36s]             class WrappedHessp:  # type:ignore
[   36s]                 __slots__ = ("fcn", "free", "par", "vec")
[   36s]     
[   36s]                 def __init__(self, fcn):
[   36s]                     self.fcn = fcn
[   36s]                     self.free = cfree
[   36s]                     self.par = cpar
[   36s]                     self.vec = np.zeros_like(self.par)
[   36s]     
[   36s]                 if self.fcn._array_call:
[   36s]     
[   36s]                     def __call__(self, par, v):
[   36s]                         self.par[self.free] = par
[   36s]                         self.vec[self.free] = v
[   36s]                         return self.fcn(self.par, self.vec)[self.free]
[   36s]     
[   36s]                 else:
[   36s]     
[   36s]                     def __call__(self, par, v):
[   36s]                         self.par[self.free] = par
[   36s]                         self.vec[self.free] = v
[   36s]                         return self.fcn(*self.par, self.vec)[self.free]
[   36s]     
[   36s]         fcn = Wrapped(self.fcn._fcn)
[   36s]     
[   36s]         grad = self.fcn._grad
[   36s]         grad = WrappedGrad(grad) if grad else None
[   36s]     
[   36s]         if hess:
[   36s]             hess = WrappedHess(hess)
[   36s]     
[   36s]         if hessp:
[   36s]             hessp = WrappedHessp(hessp)
[   36s]     
[   36s]         if constraints is not None:
[   36s]             if isinstance(constraints, dict):
[   36s]                 raise ValueError("setting constraints with dicts is not supported")
[   36s]     
[   36s]             if not isinstance(constraints, _tp.Iterable):
[   36s]                 constraints = [constraints]
[   36s]     
[   36s]             for i, c in enumerate(constraints):
[   36s]                 if isinstance(c, NonlinearConstraint):
[   36s]                     c.fun = Wrapped(c.fun)
[   36s]                 elif isinstance(c, LinearConstraint):
[   36s]                     if no_fixed_parameters == False:
[   36s]                         x = cpar.copy()
[   36s]                         x[cfree] = 0
[   36s]                         shift = np.dot(c.A, x)
[   36s]                         lb = c.lb - shift
[   36s]                         ub = c.ub - shift
[   36s] >                       A = np.atleast_1d(c.A)[:, cfree]
[   36s] E                       IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed
[
```

--- a/src/iminuit/minuit.py
+++ b/src/iminuit/minuit.py
@@ -1072,7 +1072,7 @@ class Minuit:
                         shift = np.dot(c.A, x)
                         lb = c.lb - shift
                         ub = c.ub - shift
-                        A = np.atleast_1d(c.A)[:, cfree]
+                        A = np.atleast_2d(c.A)[:, cfree]
                         constraints[i] = LinearConstraint(A, lb, ub, c.keep_feasible)
                 else:
                     raise ValueError(

